contracts/scripts/prepublish_generate_abi_folder

```diff
@@ -1,22 +1,22 @@
 #!/usr/bin/env bash
 
-# This script copies artifacts from `../artifacts/src/` and copies the production abi json files in `../abi/`
-# for publishing to the @chainlink/contracts package
+# IMPORTANT: Ensure hardhat-abi-exporter package is installed and `hardhat compile` is run before running this script.
+#
+# This script copies abi json files generated by hardhat-abi-exporter from `../abi/src/v0.x/` to `../abi/v0.x/`
+# then converts them from .json to .abi
 
 SCRIPTPATH="$( cd "$(dirname "$0")" >/dev/null 2>&1 ; cd .. && pwd -P )"
 TARGET="abi"
 
-# Delete existing
-rm -rf $SCRIPTPATH/$TARGET && mkdir $SCRIPTPATH/$TARGET
-
-# Delete debug files generated by hardhat
-find $SCRIPTPATH/artifacts/src -type f -name "*.dbg.json" -exec rm {} \;
-
-# For each version directory in src, create a subsequent directory in the target location
-# and copy all json abi files to it
+# For each version directory in src, copy the json files to the correct location
 versions=( v0.4 v0.5 v0.6 v0.7 v0.8 )
 for version in "${versions[@]}"
 do
+  rm -rf $SCRIPTPATH/$TARGET/$version
   mkdir $SCRIPTPATH/$TARGET/$version
-  find $SCRIPTPATH/artifacts/src/$version -type f -name "*.json" -exec cp {} $SCRIPTPATH/$TARGET/$version/ \;
-done
\ No newline at end of file
+  find $SCRIPTPATH/$TARGET/src/$version -type f -name "*.json" -exec cp {} $SCRIPTPATH/$TARGET/$version/ \;
+done
+
+# Remove the original src abis and rename the .json extensions to .abi
+rm -rf $SCRIPTPATH/$TARGET/src
+find $SCRIPTPATH/$TARGET -depth -type f -name "*.json" -exec sh -c 'mv "$1" "${1%.json}.abi"' _ {} \;
```

Comment:
```suggestion
find $SCRIPTPATH/$TARGET -depth -type f -name "*.json" -exec sh -c 'mv "$1" "${1%.json}.abi.json"' _ {} \;
```
On second thought, I think if it's JSON we should mark it as such. Either have the double extension, or just go with `.json`.